recipes/glib/all/conanfile.py

```diff
@@ -104,19 +109,16 @@ def validate(self):
             raise ConanInvalidConfiguration("libelf dependency can't be disabled in glib < 2.67.0")
 
     def build_requirements(self):
-        self.tool_requires("meson/0.63.2")
+        self.tool_requires("meson/0.63.3")
         self.tool_requires("pkgconf/1.9.3")
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version],
             destination=self.source_folder, strip_root=True)
 
-    def layout(self):
-        self.folders.build = self._build_subfolder
-        self.folders.source = self._source_subfolder
-
     def generate(self):
-
+        env = VirtualBuildEnv(self)
+        env.generate()
```

Comment:
I can't see it used anywhere... why is it needed?

Reply:
Because there are build requirements. My understanding is that it's implicit in conan v2 mode, but not in conan v1 mode by default.